### DISCUSSION TITLE
Correct set-up of advection term?

### DISCUSSION THREAD

**Discussion by niklasreich** [pymor/pymor#2371] [Q&A] [answered]
Hi,
I want to make some tests with the linear advection equation. More precisely, I want $f_l(x,\mu)=\mu$, where $\mu$ is a just a scalar parameter, for the built-in discretizer. But I am struggling to set up the advection term in my `StationaryProblem()`. I tried several things, for example
```
advection = LincombFunction([ConstantFunction(1., dim_domain=domain.dim)],
                            [ExpressionParameterFunctional('advection[0]', {'advection': 1})])
```
but every variant I tried resulted in `advection.shape_range` equal to `()` while `(domain.dim,)` is required. How can I achieve this? What am I missing?

Best regards,
Niklas

**Comment by sdrave** [comment, marked as answer]
Hi @niklasreich, sorry for the late reply! Like NumPy, pyMOR's functions differentiate between scalar values (`shape == ()`) and one-dimensional arrays of length 1 (`shape == (1,)`). `advection` needs to be a vector field. In 1d, this means `shape_range` needs to be `(1,)`.  So you need to set the value of `ConstantFunction` to be a 1d array instead of a scalar:

```python
 advection = LincombFunction([ConstantFunction(np.array([1.]), dim_domain=1)],
                             [ExpressionParameterFunctional('advection[0]', {'advection': 1})])
```

Does that work for you?

**Comment by niklasreich** [comment]
Hi @sdrave, thanks for the explanation and the code. It works just fine. Somehow I was not able to come up with this solution.
